Library/Homebrew/utils/info.rb

```diff
@@ -0,0 +1,24 @@
+# typed: true
+# frozen_string_literal: true
+
+module Utils
+  # Helper functions for formula and cask info.
+  module Info
+    def self.decorate_dependencies(dependencies)
```

Comment:
Looking at the implementation here: as you've mentioned formulae and casks don't have enough overlap here for the implementation to be sharable between formulae/casks.

Instead, I'd suggest adding a `decorate_dependencies` method to `Library/Homebrew/cask/info.rb` which takes the formula/casks as strings and then uses that to calculate dependencies, relying on `pretty_installed`/`pretty_installed` for the shared logic instead.

Make sense? 

Shout if you want more help in that direction.